bmcremedy_connector.py

```diff
@@ -564,23 +706,198 @@ def _make_rest_call(self, endpoint, action_result, headers=None, params=None, da
+
+        phantom_base_url = resp_json.get('base_url')
+        if not phantom_base_url:
+            return action_result.set_status(
+                phantom.APP_ERROR, "Phantom Base URL is not configured, please configure it in System Settings"), None
+
+        phantom_base_url = phantom_base_url.strip("/")
+
+        return phantom.APP_SUCCESS, phantom_base_url
+
+    def _get_asset_name(self, action_result):
+
+        ret_val, resp_json = self._make_rest_calls_to_phantom(
+            action_result, '{}rest/asset/{}'.format(self.get_phantom_base_url(), self.get_asset_id()))
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), None
+
+        asset_name = resp_json.get('name')
+        if not asset_name:
+            return action_result.set_status(phantom.APP_ERROR, "Error retrieving asset name"), None
+
+        return phantom.APP_SUCCESS, asset_name
+
+    def _get_url_to_app_rest(self, action_result=None):
+        if not action_result:
+            action_result = ActionResult()
+        # get the phantom ip to redirect to
+        ret_val, phantom_base_url = self._get_phantom_base_url_bmc(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), action_result.get_message()
+        # get the asset name
+        ret_val, asset_name = self._get_asset_name(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), action_result.get_message()
+        self.save_progress('Using Phantom base URL as: {0}'.format(phantom_base_url))
+        app_json = self.get_app_json()
+        app_name = app_json['name']
+        app_dir_name = _get_dir_name_from_app_name(app_name)
+        url_to_app_rest = "{0}/rest/handler/{1}_{2}/{3}".format(phantom_base_url, app_dir_name, app_json['appid'], asset_name)
+        return phantom.APP_SUCCESS, url_to_app_rest
+
+    def _bmc_int_auth_initial(self, client_id, client_secret):
+
+        self.debug_print("In bmc_int_auth_initial method")
+
+        state = self.rsh.load_state()
+        asset_id = self.get_asset_id()
+
+        ret_val, message = self._get_url_to_app_rest()
+        if phantom.is_fail(ret_val):
+            return phantom.APP_ERROR, message
 
-        # See if an error message is present
-        message = response_data.get('message', consts.BMCREMEDY_REST_RESP_OTHER_ERROR_MESSAGE)
-        error_message = consts.BMCREMEDY_ERROR_FROM_SERVER.format(status=response.status_code, detail=message)
-        self.debug_print(error_message)
+        app_rest_url = message
 
-        # Set the action_result status to error, the handler function will most probably return as is
-        return RetVal3(action_result.set_status(phantom.APP_ERROR, error_message), response_data, response)
+        request_url = '{}/rsso/oauth2'.format(self._base_url)
+
+        proxy = {}
+        if 'HTTP_PROXY' in os.environ:
+            proxy['http'] = os.environ.get('HTTP_PROXY')
```

Comment:
No need for `get` as we already checked this key is present, same below

Reply:
Removed the get method.